source/_integrations/google_assistant_sdk.markdown

```diff
@@ -69,13 +69,19 @@ The integration setup will next give you instructions to enter the [Application
 
 {% enddetails %}
 
+## IPv6
+
+Due to limitations in Google's SDK, broadcasts do not currently support IPv6. In order for broadcasts to work, IPv6 needs to be disabled for both your Home Assistant server as well as all speakers. To disable IPv6 on your Home Assistant server, set `net.ipv6.conf.all.disable_ipv6=1` in `/etc/sysctl.conf` or your Docker configuration.
```

Comment:
There is no such thing as broacasts in IPv6.

As we don't have sysctl configuration support in Home Assistant OS, disabling IPv6 that way is not possible.

What is the problem here exactly?

Reply:
@agners I used that sysctl for my Docker container and it works well there.  I haven't used HAOS though, but does it not support sysctls at all? I thought it was Linux based?

The problem is that the docs currently say to disable IPv6 across the entire network, when really you *only* need to disable IPv6 on the Home Assistant server and the Google speakers. Everything else on the network can continue to use it.

Disabling IPv6 for the entire network is a bad idea in 2023, especially as many ISPs outside the USA are using CGNAT for IPv4. In the case of CGNAT, IPv6 is the only way to make systems externally accessible (for the cases where you can't use a VPN).

Reply:
HAOS is Linux based and uses sysctl. However, it is largely statelesss and much of the rootfs is read-only, including `/etc/sysctl.conf`. So that won't work on HAOS as-is.

> Disabling IPv6 for the entire network is a bad idea in 2023, especially as many ISPs outside the USA are using CGNAT for IPv4. In the case of CGNAT, IPv6 is the only way to make systems externally accessible (for the cases where you can't use a VPN).

100% agree, in fact with Thread and Matter based on IPv6, I'd say one should not disable IPv6 on Home Asssistant even :smile: 

I'd like to understand what the issue here is at all. In a quick try in my home, where all devices have global and ULA and link-local IPv6 addresses, asking the speaker t broadcast things worked just fine. Is it only a HA issue? :thinking: Are there Core issues I could read up about this?

Reply:
Marking this as *Draft* as there is an ongoing discussion. Please undraft it once you feel the discussion has been resolved.

Reply:
>  Are there Core issues I could read up about this?

@agners Sorry for the delay in replying.  It's an issue with the Google Assistant SDK rather than anything specific to Home Assistant.

Please see prior discussions:
 - https://greghesp.github.io/assistant-relay/docs/other/faq/#i-tried-a-broadcast-but-i-dont-hear-anything (from a different software product)
 - https://stackoverflow.com/a/62772434 (the answer here lacks substance though)
 - https://community.home-assistant.io/t/google-assistant-sdk-notify-doesnt-work/514036/2
 - https://github.com/home-assistant/core/issues/84726

Google have deprecated the Python Google Asssistant SDK and I doubt they'll ever fix this issue. It seems like they don't care about development of Google Assistant any more.

Reply:
Ok, thanks for these links!

I've also reproduced this on my end, and I can confirm the same observation.

It doesn't seem necessary to disable IPv6 entirely using sysctl. Disabling the SLAAC/DHCPv6 to assign ULA/global addresses seems sufficient. Or in other words: Google Assistant broadcast works even with link-local IPv6 addressing enabled. That is exactly what the "Disabled" option of the Home Assistant OS network configuration does:

![image](https://github.com/home-assistant/home-assistant.io/assets/34061/b060e800-a888-45a0-8680-b12d0252bb6d)

Can you adjust the text to suggest this method on Home Assistant OS installations?

Also, I think it makes sense to link the Core issue here to make it clear that this is a recognized problem we can't fix.

Reply:
Maybe make the title a bit more verbose, e.g. "IPv6 limitations when using Broadcast feature" or "Broadcast feature and IPv6". :thinking: